app/web/features/search/SearchBox.tsx

```diff
@@ -40,186 +29,104 @@ const useStyles = makeStyles((theme) => ({
-                    searchFilters.active.lat ?? 0
-                  ),
-                  bbox: searchFilters.active.bbox ?? [0, 0, 0, 0],
-                }
-              : ""
-          }
+          defaultValue={locationResult.name}
           label={t("search:form.location_field_label")}
-          onChange={handleNewLocation}
+          onChange={(event) => {
+            if (event) {
+              setLocationResult(event);
+            }
+          }}
           fieldError={errors.location?.message}
           disableRegions
         />
       ) : (
-        <Controller
-          control={control}
-          name="query"
-          defaultValue={searchFilters.active.query ?? ""}
-          render={({ value, onChange }) => (
-            <TextField
-              fullWidth
-              id="query"
-              value={value}
-              label={t("search:form.keywords.field_label")}
-              variant="standard"
-              helperText=" "
-              onChange={(event) => {
-                onChange(event.target.value);
-                handleKeywordsChangeDebounced(event);
-              }}
-              InputProps={{
-                endAdornment: (
-                  <InputAdornment position="end">
-                    <IconButton
-                      aria-label={t(
-                        "search:form.keywords.clear_field_action_a11y_label"
-                      )}
-                      onClick={() => {
-                        setValue("query", "");
-                        handleKeywordsChange("");
-                      }}
-                      size="small"
-                    >
-                      <CrossIcon />
-                    </IconButton>
-                  </InputAdornment>
-                ),
-              }}
-            />
-          )}
+        <TextField
+          fullWidth
+          id="query"
+          value={queryName}
+          label={t("search:form.keywords.field_label")}
+          variant="standard"
+          helperText=" "
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            setQueryName(event.target.value);
+            setLocationResult({
+              ...locationResult,
+              location: undefined,
+            });
+          }}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label={t(
+                    "search:form.keywords.clear_field_action_a11y_label"
+                  )}
+                  onClick={() => {
+                    setQueryName("");
+                    setLocationResult({
+                      ...locationResult,
+                      location: undefined,
+                    });
+                  }}
+                  size="small"
+                >
+                  <CrossIcon />
+                </IconButton>
+              </InputAdornment>
+            ),
+          }}
```

Comment:
If I understand correctly, you're passing a child component through the props. 

Would it not be better to make this a separate component and just export it into the `TextField` component? 

Or is there reasoning here I'm missing?

Reply:
Sure I won't fight you on it haha, just makes things more readable and shorter I think. Also if you ever wanna write tests for that component itself.